python/data_model.py

```diff
@@ -0,0 +1,122 @@
+    'dataset': {
+        'pubMedId': 'Dataset Pubmed Id',
+        'grantNumber': 'Dataset Grant Number',
+        'consortium': 'Dataset Consortium Name',
+        'theme': 'Dataset Theme Name',
+        'datasetName': 'Dataset Name',
+        'datasetAlias': 'Dataset Alias',
+        'description': 'Dataset Description',
+        'overallDesign': 'Dataset Design',
+        'assay': 'Dataset Assay',
+        'species': 'Dataset Species',
+        'tumorType': 'Dataset Tumor Type',
+        'externalLink': 'Dataset External Link'
+    },
+    'tool': {
+        'pubMedId': 'Tool Pubmed Id',
+        'consortium': 'Tool Consortium Name',
+        'grantNumber': 'Tool Grant Number',
+        'toolName': 'Tool Name',
+        'description': 'Tool Description',
+        'homepage': 'Tool Homepage',
+        'version': 'Tool Version',
+        'operation': 'Tool Operation',
+        'inputData': 'Tool Input Data',
+        'outputData': 'Tool Output Data',
+        'inputFormat': 'Tool Input Format',
+        'outputFormat': 'Tool Output Format',
+        'functionNote': 'Tool Function Note',
+        'cmd': 'Tool Cmd',
+        'toolType': 'Tool Type',
+        'topic': 'Tool Topic',
+        'operatingSystem': 'Tool Operating System',
+        'language': 'Tool Language',
+        'license': 'Tool License',
+        'cost': 'Tool Cost',
+        'accessibility': 'Tool Accessibility',
+        'downloadUrl': 'Tool Download Url',
+        'downloadType': 'Tool Download Type',
+        'downloadNote': 'Tool Download Note',
+        'downloadVersion': 'Tool Download Version',
+        'documentationUrl': 'Tool Documentation Url',
+        'documentationType': 'Tool Documentation Type',
+        'documentationNote': 'Tool Documentation Note',
+        'linkUrl': 'Tool Link Url',
+        'linkType': 'Tool Link Type',
+        'linkNote': 'Tool Link Note'
+    },
+    'project': {
+        'projectName': 'Project Name',
+        'projectType': 'Project Type',
+        'description': 'Project Description',
+        'theme': 'Project Theme Name',
+        'consortium': 'Project Consortium Name',
+        'grant': 'Project Grant Number'
+    }
+}
+
+# Extra columns in the table schema that do not exist in the data model.
+DROP = {
+    'publication': [
+        'publicationId', 'pubMedLink', 'themeId', 'consortiumId',
+        'grantId', 'grantName', 'grantInstitution', 'datasetId'
+    ],
+    'dataset': [
+        'datasetId', 'themeId', 'consortiumId', 'grantId', 'grantName',
+        'publicationId', 'publicationTitle', 'publication'
+    ],
+    'tool': [
+        'toolId', 'grantId', 'grantName', 'publicationTitle',
+        'downloadLink', 'portalDisplay'
+    ],
+    'project': [
+        'projectId', 'themeId', 'consortiumId', 'grantId', 'grantName', 'grantType'
+    ]
+}
+
+# Missing columns in the table schema that exist in the data model.
+ADD = {
+    'publication': [
+        {'index': 15, 'colname': 'Publication Tool Name', 'value': ""},
```

Comment:
I took out Publication Tool Name from the data model, so I think we can remove this? Unless it does not make a difference to be in here.

Reply:
Ah right, good catch!